{"bug_class": "PROVER", "kind": "ERROR", "bug_type": "THREAD_SAFETY_VIOLATION", "qualifier": "Unprotected write. Non-private method `org.jboss.xnio.nio.NioProvider.start` writes to field `this.org.jboss.xnio.nio.NioProvider.executorService` outside of synchronization.\n Reporting because another access to the same memory occurs on a background thread, although this access may not.", "severity": "HIGH", "visibility": "user", "line": 116, "column": -1, "procedure": "void NioProvider.start()", "procedure_id": "org.jboss.xnio.nio.NioProvider.start():void.f47e39311567aba21f50d71b61677d30", "procedure_start_line": 111, "file": "nio-impl/src/main/java/org/jboss/xnio/nio/NioProvider.java", "bug_trace": [{"level": 0, "filename": "nio-impl/src/main/java/org/jboss/xnio/nio/NioProvider.java", "line_number": 116, "column_number": -1, "description": "<Write on unknown thread>"}, {"level": 0, "filename": "nio-impl/src/main/java/org/jboss/xnio/nio/NioProvider.java", "line_number": 116, "column_number": -1, "description": "access to `this.org.jboss.xnio.nio.NioProvider.executorService`"}, {"level": 0, "filename": "nio-impl/src/main/java/org/jboss/xnio/nio/NioProvider.java", "line_number": 170, "column_number": -1, "description": "<Write on background thread>"}, {"level": 0, "filename": "nio-impl/src/main/java/org/jboss/xnio/nio/NioProvider.java", "line_number": 170, "column_number": -1, "description": "access to `this.org.jboss.xnio.nio.NioProvider.executorService`"}], "key": "NioProvider.java|start|THREAD_SAFETY_VIOLATION", "node_key": "9c5d6d9028928346cc4fb44cced5dea1", "hash": "2ae8eca9e38e5f9567123bd26bf7915f", "bug_type_hum": "Thread Safety Violation", "censored_reason": ""}